packages/atlas-service/src/main.ts

```diff
@@ -524,6 +533,64 @@ export class AtlasService {
     await throwIfNotOk(res);
   }
 
+  static async getAIFeatureEnablement(): Promise<AIFeatureEnablement> {
+    throwIfNetworkTrafficDisabled();
+
+    const userId = (await this.getActiveCompassUser()).id;
+
+    const res = await this.fetch(
+      `${this.config.atlasApiBaseUrl}/ai/api/v1/hello/${userId}`
+    );
+
+    await throwIfNotOk(res);
+
+    const body = await res.json();
+
+    validateAIFeatureEnablementResponse(body);
+
+    return body;
+  }
+
+  static async setupAIAccess(): Promise<void> {
+    log.info(
+      mongoLogId(1_001_000_227),
+      'AtlasService',
+      'Fetching if the AI feature is enabled'
+    );
+
+    try {
+      throwIfNetworkTrafficDisabled();
+
+      const featureResponse = await this.getAIFeatureEnablement();
+
+      const isAIFeatureEnabled =
+        !!featureResponse.features.GEN_AI_COMPASS?.enabled;
+
+      log.info(
+        mongoLogId(1_001_000_229),
+        'AtlasService',
+        'Fetched if the AI feature is enabled',
+        {
+          enabled: isAIFeatureEnabled,
+        }
+      );
+
+      await preferences.savePreferences({
+        cloudFeatureRolloutAccess: {
+          GEN_AI_COMPASS: isAIFeatureEnabled,
+        },
+      });
+    } catch (err) {
+      // Default to what's already in Compass when we can't fetch the preference.
+      log.error(
+        mongoLogId(1_001_000_243),
+        'AtlasService',
+        'Failed to load if the AI feature is enabled',
+        { error: (err as Error).stack }
+      );
```

Comment:
@mcasimir and I were chatting on this. When someone is not connected to the internet when they first open Compass we still want them to be able to use the feature. If we set it to disabled then they can't use it, even if they've used it in the past.
It makes a case for having some polling or something here.

Reply:
Doesn't seem like it's a good way to work around a corner case. In combination with storing this value with preferences on disk this error handling creates a very easy way to bypass the whole check. Feel free to ignore it if that's what we already decided on

Reply:
I think we are assuming these requirements here:

- the ping should be completely transparent for the end user, failures should never affect the usage in any way 
- the last value "read" should be used by Compass: the purpose of the flag is to allow for a progressive rollout to % of users and for a rollback of a % of the usage (in extreme cases). 
- we don't need a security mechanism, is fine if someone finds ways to bypass it
- the feature should not be disabled intermittently (due to a failed ping), once is enabled for a user, it should stay like that unless is intentionally rolled back

Is there anything that is not matching here?

Reply:
> the feature should not be disabled intermittently (due to a failed ping), once is enabled for a user, it should stay like that unless is intentionally rolled back

What is "rolled back" in this case? If backend returned feature as enabled initially, can it return it as disabled later? If the answer is "yes", then my concern still applies, but it seems like we are okay with all the implications of that. If the answer is "no", then we probably shouldn't try to fetch the flags again if the flags are not stored yet, feature was disabled before.

Reply:
Yes, the backend can return the feature as enabled initially, and then return it as disabled later.
My impression is that we're okay with the implications there. @mcasimir is that alright with you?
Ultimately we can manage access and request load through the backend.

Reply:
Yeah, that's to keep the possibility to scale down. Would be good to double check if we really want that though. 

> then my concern still applies

Do you refer to this? "is it okay that if the feature is disabled now and we somehow failed to fetch it, it will not be disabled if it was enabled before?"

I guess, wouldn't it be bad if for a sporadic failure the feature is going to get disabled?

Reply:
It wouldn't be great, but if we want this to be a reliable killswitch for the feature, we should probably not ignore errors and have another pass as designing how we want to handle that. If we see this as a simple and temporary rollout mechanism and just generally okay with people escaping it, then it's probably fine to just ignore this

Reply:
That would be great to make it clearer in the code what's the purpose of this then, yes 👍